apps/portal/src/utils/errors.js

```diff
@@ -9,24 +9,110 @@ export class HumanReadableError extends Error {
         if (res.status === 400 || res.status === 429) {
             try {
                 const json = await res.json();
-                if (json.errors && Array.isArray(json.errors) && json.errors.length > 0 && json.errors[0].message) {
-                    return new HumanReadableError(json.errors[0].message);
+                if (json.errors && Array.isArray(json.errors) && json.errors.length > 0 && json.errors[ 0 ].message) {
+                    return new HumanReadableError(json.errors[ 0 ].message);
                 }
             } catch (e) {
                 // Failed to decode: ignore
                 return false;
             }
         }
+        if (res.status === 500) {
+            return new HumanReadableError('A server error occurred');
+        }
     }
+}
 
-    /**
-     * Only output the message of an error if it is a human readable error and should be exposed to the user.
-     * Otherwise it returns a default generic message.
-     */
-    static getMessageFromError(error, defaultMessage) {
-        if (error instanceof HumanReadableError) {
-            return error.message;
+// changing this - will export this helper.  It's not helping me to have it stuck to HumanReadableError... 
```

Comment:
This could likely be cleaned up.

Reply:
Yeah, sorry. Bit of a note to myself, followed by a now-inaccurate comment.  Will fix!